invoices/invoiceregistry.go

```diff
@@ -62,6 +62,14 @@ type InvoiceRegistry struct {
 	// value from the payment request.
 	decodeFinalCltvExpiry func(invoice string) (uint32, error)
 
+	// subscriptions is a map from a payment hash to a list of subscribers.
+	// It is used for efficient notification of links.
+	hodlSubscriptions map[lntypes.Hash]map[chan<- interface{}]struct{}
+
+	// reverseSubscriptions tracks hashes subscribed to per subscriber. This
+	// is used to unsubscribe from all hashes efficiently.
+	hodlReverseSubscriptions map[chan<- interface{}]map[lntypes.Hash]struct{}
```

Comment:
Using a map as a key here makes be uneasy. Above as well, can we instead use a monotonic client counter?

Reply:
Also why `interface{}` if we already know the type? We may get one from the queue itself, but doesn't mean we need to sacrifice type safety everywhere.

Reply:
I don't use a map as a key. I use a chan as a key.

Assuming you meant to say using a chan as a key makes you uneasy: why is that? I'd agree if the key was a struct. Those can be copied accidentally and thereby introduce bugs. A channel cannot be copied (?), so I think this is safe to do.

Reply:
W/r `interface{}`, how would that work if I'd put `chan <-HodlEvent` here? I cannot just connect channels of different types without another goroutine just for the conversion.